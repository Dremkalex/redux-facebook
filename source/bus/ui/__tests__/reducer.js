// Reducer
import { uiReducer } from "../reducer";

// Actions
import { uiActions } from "../actions";

describe("ui reducer:", () => {
    test("should return initial state by default", () => {
        expect(uiReducer(void 0, {})).toMatchSnapshot();
    });

    test("should handle a START_FETCHING action", () => {
        expect(uiReducer(void 0, uiActions.startFetching())).toMatchSnapshot();
    });

    test("should handle a STOP_FETCHING action", () => {
        expect(
            uiReducer(void 0, uiActions.stopFetching())).toMatchSnapshot();
    });

    test("should handle a SET_ONLINE_STATE action", () => {
        expect(uiReducer(void 0, uiActions.setOnlineState())).toMatchSnapshot();
    });

    test("should handle a SET_OFFLINE_STATE action", () => {
        expect(uiReducer(void 0, uiActions.setOfflineState())).toMatchSnapshot();
    });
});
