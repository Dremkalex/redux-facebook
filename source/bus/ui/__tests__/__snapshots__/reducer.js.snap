// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui reducer: should handle a SET_OFFLINE_STATE action 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`ui reducer: should handle a SET_ONLINE_STATE action 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": true,
}
`;

exports[`ui reducer: should handle a START_FETCHING action 1`] = `
Immutable.Map {
  "isFetching": true,
  "isOnline": false,
}
`;

exports[`ui reducer: should handle a STOP_FETCHING action 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`ui reducer: should return initial state by default 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;
